### PR TITLE
Refactor and Perfomance Improvment in bot_jobs

### PR DESCRIPTION
**Pull request by noppakorn**
- Refactor utils.py
- Use numpy to calculate moving_average

### CODE REVIEW

File: bot_jobs/util.py [hunk 1]
```diff
@@ -1,45 +1,36 @@
...
-    return pdata
+        json_data = json.load(json_file)
+    return tuple(province["properties"]["PROV_NAMT"] for province in json_data["features"])
+
 
 def get_start_end(data):
-    lastRow = data.tail(1)
-    date = list(lastRow['announce_date'])[0].strip()
     start = datetime.datetime.strptime("2020-12-15", "%Y-%m-%d")
+    date = data.tail(1)['announce_date'].iloc[0].strip()
```
Comment (plumsirawit) [inline bot_jobs/util.py:21]: nice

File: components/gis/data/manufacturer-vaccination-data.json [hunk 2]
```diff
@@ -1 +1 @@
...
...es","doses_administered":86072},{"date":"2021-06-10","manufacturer":"Sinovac Life Sciences","doses_administered":98494},{"date":"2021-06-10","manufacturer":"AstraZeneca","doses_administered":217883},{"date":"2021-06-11","manufacturer":"AstraZeneca","doses_administered":218119},{"date":"2021-06-11","manufacturer":"Sinovac Life Sciences","doses_administered":169921},{"date":"2021-06-12","manufacturer":"Sinovac Life Sciences","doses_administered":65937},{"date":"2021-06-12","manufacturer":"AstraZeneca","doses_administered":70262},{"date":"2021-06-13","manufacturer":"AstraZeneca","doses_administered":46385},{"date":"2021-06-13","manufacturer":"Sinovac Life Sciences","doses_administered":58746},{"date":"2021-06-14","manufacturer":"AstraZeneca","doses_administered":128939},{"date":"2021-06-14","manufacturer":"Sinovac Life Sciences","doses_administered":160304},{"date":"2021-06-15","manufacturer":"Sinovac Life Sciences","doses_administered":148057},{"date":"2021-06-15","manufacturer":"AstraZeneca","doses_administered":92771},{"date":"2021-06-16","manufacturer":"AstraZeneca","doses_administered":46826},{"date":"2021-06-16","manufacturer":"Sinovac Life Sciences","doses_administered":52551}]
```
Comment (plumsirawit) [inline components/gis/data/manufacturer-vaccination-data.json]: should not update

File: components/gis/data/provincial-vaccination-data.json [hunk 3]
```diff
@@ -1 +1 @@
...
..."total_doses":23756,"total-1st-dose":15937,"total-2nd-dose":7819,">60-total-doses":3240},{"name":"ตรัง","id":"92","population":640574,"registered_population":640574,">60-population":105627,"coverage":0.025299965343582476,"total_doses":32413,"total-1st-dose":24249,"total-2nd-dose":8164,">60-total-doses":7710},{"name":"พัทลุง","id":"93","population":523077,"registered_population":523077,">60-population":100108,"coverage":0.024093011162792475,"total_doses":25205,"total-1st-dose":18973,"total-2nd-dose":6232,">60-total-doses":7697},{"name":"ปัตตานี","id":"94","population":726015,"registered_population":726015,">60-population":88552,"coverage":0.024326632369854616,"total_doses":35323,"total-1st-dose":28181,"total-2nd-dose":7142,">60-total-doses":5563},{"name":"ยะลา","id":"95","population":538602,"registered_population":538602,">60-population":65546,"coverage":0.02436214496047174,"total_doses":26243,"total-1st-dose":19424,"total-2nd-dose":6819,">60-total-doses":3849},{"name":"นราธิวาส","id":"96","population":804429,"registered_population":804429,">60-population":94169,"coverage":0.03105308237271406,"total_doses":49960,"total-1st-dose":39707,"total-2nd-dose":10253,">60-total-doses":8444}]}
```
Comment (plumsirawit) [inline components/gis/data/provincial-vaccination-data.json]: should not update

File: bot_jobs/util.py [hunk 4]
```diff
@@ -1,45 +1,36 @@
...
-    for i, x in enumerate(ys, 1):
-        cumulative_sum.append(cumulative_sum[i - 1] + x)
-        if i >= N:
-            moving_ave = (cumulative_sum[i] - cumulative_sum[i - N]) / N
-            # can do stuff with moving_ave here
-            moving_aves.append(moving_ave)
-    return moving_aves
\ No newline at end of file
+    if len(ys) < N : return []
+    return np.convolve(np.array(ys), np.ones(N), "valid") / N
```
Comment (plumsirawit) [inline bot_jobs/util.py]: please add endline

**Review by plumsirawit — CHANGES_REQUESTED** [review]
Looks good, please fix according to comments. Thanks.

File: components/gis/data/manufacturer-vaccination-data.json (hunk 2 above)
Comment (noppakorn) [reply components/gis/data/manufacturer-vaccination-data.json]: Fixed, Thanks for your service.

File: components/gis/data/provincial-vaccination-data.json (hunk 3 above)
Comment (noppakorn) [reply components/gis/data/provincial-vaccination-data.json]: Fixed, Thanks for your service.

File: bot_jobs/util.py (hunk 4 above)
Comment (noppakorn) [reply bot_jobs/util.py]: Fixed, lol

Comment (plumsirawit) [reply bot_jobs/util.py]: I don't think so.

**Review by plumsirawit — APPROVED** [review]
Looks good to me!

**Pull request merged**